### PR TITLE
Bug fix

### PR DESCRIPTION
**Pull request by anjalidk**
For CPU's all fields may not be present in some OSes. Fix that by checking if every field is present before printing.

### CODE REVIEW

File: rescpu.c [hunk 1]
```diff
@@ -29,9 +29,7 @@
 #include <libgen.h>
 #include <ctype.h>
 
-//#define TESTING
-
-static int populate_cpuinfo(void *out, char *buffer, int total_len)
+static int populate_cpuinfo(void *out, char *buffer, int total_len, int ex)
```
Comment (hallyn) [inline rescpu.c:32]: maybe rename 'ex' to 'probe' or something (and invert it)?  And then add a comment above explaining that if probe is true then you only set the value to 1 to say it was present.

**Review by hallyn — APPROVED** [review]
Ok, but a suggestion for clarity.

**Comment by anjalidk** [comment]
> On Aug 30, 2023, at 8:08 AM, Serge Hallyn ***@***.***> wrote:
> 
> 
> @hallyn approved this pull request.
> 
> Ok, but a suggestion for clarity.
> 
> In rescpu.c:
> 
> > @@ -29,9 +29,7 @@
>  #include <libgen.h>
>  #include <ctype.h>
>  
> -//#define TESTING
> -
> -static int populate_cpuinfo(void *out, char *buffer, int total_len)
> +static int populate_cpuinfo(void *out, char *buffer, int total_len, int ex)
> 
> maybe rename 'ex' to 'probe' or something (and invert it)? And then add a comment above explaining that if probe is true then you only set the value to 1 to say it was present.
> 
Will do, not sure what you mean by invert it?

**Comment by hallyn** [comment]
> Will do, not sure what you mean by invert it?

Actually I was thinking wrongly.  'ex' and 'probe' are both true if you only want to set the field to 1 to indicate its presence.  So ignore that part of what I said.

**Comment by anjalidk** [comment]
Thanks, review comments done. Submitted in same pull #29.
Anjali

**Comment by hallyn** [comment]
Hm, it would also help if to have meaningful consts like CPUINFO_PROBEONLY 1 and CPUINFO_FULL 0 or something, but let's merge for now, thanks.

**Comment by anjalidk** [comment]
Hi Serge,
For some reason I am unable to create new pull requests - I tried this for the branch “rev1” but I do not see any “create pull request” handle - any idea what is going wrong?

Thanks
Anjali

**Comment by anjalidk** [comment]
Ok, will do.
Anjali

**Pull request merged**